### PR TITLE
Adding PR template

### LINKED ISSUE #165: [FEATURE] Add a PR template
### Detailed description

Adding a pull request template will make the work for maintainers much easier.

### Context

This would make the review process easier, plus, letting the maintainers/reviewers know what the PR is about, useful information, and more.

### Possible implementation

_No response_

### Additional information

If you wanna work on this, feel free to raise a PR. Contributions are always welcome!

### PR DESCRIPTION
**Pull request by MukundLadani**
Adding PR template after making all required changes 

Closes #165 .

## Changes

-Added PR template
-Also showed an Example of the template with binary search function in the template itself.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<!-- Follow this template for PR's -->
```
Comment (appgurueu) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: Useless comment

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+<!--Example Template for adding feature of binary search function
```
Comment (appgurueu) [inline .github/PULL_REQUEST_TEMPLATE.md:3]: Useless comment

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ -->
+
+<!-- Short description of your changes -->
+
+Adding code snippet of binary search function in algorithms directory
```
Comment (appgurueu) [inline .github/PULL_REQUEST_TEMPLATE.md:8]: ???

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+<!-- Add issue no. which corresponds to this PR -->
```
Comment (appgurueu) [inline .github/PULL_REQUEST_TEMPLATE.md:10]: *if applicable*; most PRs will be adding algorithms and we don't allow issues for missing algorithms

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+Closes #no.
+
+## Changes
+
+<!-- Go verbose here & explain in bit detailed way what you have changed -->
+
+-added binary search in algorithms of code snippet
```
Comment (appgurueu) [inline .github/PULL_REQUEST_TEMPLATE.md:18]: Bad formatting, full of redundant information, binary search? This should be part of the comment if anything.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+Adding code snippet of binary search function in algorithms directory
+
+<!-- Add issue no. which corresponds to this PR -->
+
+Closes #no.
+
+## Changes
+
+<!-- Go verbose here & explain in bit detailed way what you have changed -->
```
Comment (appgurueu) [inline .github/PULL_REQUEST_TEMPLATE.md:16]: "bit detailed"? Detailed or brief?

**Comment by MukundLadani** [comment]
Hello @appgurueu, I have made the changes in the template please review it and let me know if there are any further changes.

**Pull request closed without merging**